src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -109,19 +109,23 @@ public DataWriter(HdfsSinkConnectorConfig connectorConfig, SinkTaskContext conte
               + "the path to the keytab of the principal.");
         }
 
+        conf.set("hadoop.security.authentication", "kerberos");
+        conf.set("hadoop.security.authorization", "true");
         String hostname = InetAddress.getLocalHost().getCanonicalHostName();
         // replace the _HOST specified in the principal config to the actual host
         String principal = SecurityUtil.getServerPrincipal(principalConfig, hostname);
-        String namenodePrincipalConfig = connectorConfig.getString(HdfsSinkConnectorConfig.CONNECT_HDFS_PRINCIPAL_CONFIG);
-        String namenodePrincipal = SecurityUtil.getServerPrincipal(namenodePrincipalConfig, hostname);
+        String namenodePrincipalConfig = connectorConfig.getString(HdfsSinkConnectorConfig.HDFS_NAMENODE_PRINCIPAL_CONFIG);
 
+        String namenodePrincipal = SecurityUtil.getServerPrincipal(namenodePrincipalConfig, hostname);
         // namenode principal is needed for multi-node hadoop cluster
         if (conf.get("dfs.namenode.kerberos.principal") == null) {
           conf.set("dfs.namenode.kerberos.principal", namenodePrincipal);
         }
+        log.info("Hadoop namenode principal: " + conf.get("dfs.namenode.kerberos.principal"));
 
-        final UserGroupInformation ugi = UserGroupInformation.loginUserFromKeytabAndReturnUGI(principal,
-                                                                                              keytab);
+        UserGroupInformation.setConfiguration(conf);
+        UserGroupInformation.loginUserFromKeytab(principal, keytab);
+        final UserGroupInformation ugi = UserGroupInformation.getLoginUser();
```

Comment:
I'm a bit confused because you mentioned Hadoop 2.5.0, and I can't find these methods in the API (they exist in the older 1.0.x APIs).

It may be an Hadoop documentation issue? 
http://hadoop.apache.org/docs/r2.5.2/api/